### PR TITLE
Update GMT Dev Tests workflow to test on macOS-12 and ubuntu-22.04

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Look GitHub Actions has some public beta runners available:
- macOS-12.0 (Monterey) as of 27 Apr 2022 (https://github.com/actions/virtual-environments/issues/5446)
- ubuntu-22.04 (Jammy Jellyfish) as of 5 May 2022 (https://github.com/actions/virtual-environments/issues/5490)

So let's test it out on the GMT Dev Tests workflow!

~~Also updating the cron tab to run GMT Dev Tests on [Monday/Wednesday/Friday](https://stackoverflow.com/questions/31260837/how-to-run-a-cron-job-on-every-monday-wednesday-and-friday) only. A bit annoying to have the tests failing everyday, especially during the weekend. See also https://github.com/GenericMappingTools/pygmt/pull/1833#issuecomment-1073488403 about having less frequent tests.~~ Edit: cherry-picked this to #1922 

Supersedes #1105.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
It seems that the macOS-12.0 agent doesn't respond:
https://github.com/GenericMappingTools/pygmt/runs/6315841035?check_suite_focus=true

**Comment by weiji14** [comment]
Interesting. I saw that the macos-12.0 didn't repond either when in PR at https://github.com/google/santa/pull/798, but their macos-12.0 CI worked in the main branch after merging. Maybe one of those permissions thing again. Do you think we shoud just merge and see?

**Comment by seisman** [comment]
> Interesting. I saw that the macos-12.0 didn't repond either when in PR at [google/santa#798](https://github.com/google/santa/pull/798), but their macos-12.0 CI worked in the main branch after merging. Maybe one of those permissions thing again. Do you think we shoud just merge and see?

I've applied your changes to the main branch of my own fork (https://github.com/seisman/pygmt). The macOS-12.0 runner still doesn't respond (https://github.com/seisman/pygmt/runs/6332825323?check_suite_focus=true).

**Comment by weiji14** [comment]
Ok, let's wait until next week and try again.

**Comment by seisman** [comment]
Perhaps we can pick the commits fdd64acd934a221ddb84c1221c060125e0229bda and 6b9c5ab9be7ef4a3e4da826f6a3d4b4c2c791967 into #1833, so that we run full tests 3 times a week, then #1833 may be OK to merge?

**Comment by weiji14** [comment]
> Perhaps we can pick the commits [fdd64ac](https://github.com/GenericMappingTools/pygmt/commit/fdd64acd934a221ddb84c1221c060125e0229bda) and [6b9c5ab](https://github.com/GenericMappingTools/pygmt/commit/6b9c5ab9be7ef4a3e4da826f6a3d4b4c2c791967) into #1833, so that we run full tests 3 times a week, then #1833 may be OK to merge?

I've cherry-picked the Mon/Wed/Fri crontab to #1922. Would prefer the full tests in #1833 to run on a less frequent schedule, but let's discuss that in #1833.

**Comment by weiji14** [comment]
Testing on Ubuntu 22.04 (Jammy Jellyfish) at https://github.com/GenericMappingTools/pygmt/runs/6389148872?check_suite_focus=true, seems to be running. The macOS-12.0 still seems stuck though.

File: .github/workflows/ci_tests_dev.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ['3.10']
-        os: [ubuntu-latest, macOS-11.0, windows-2022]
+        os: [ubuntu-22.04, macOS-12.0, windows-2022]
```
Comment (weiji14) [inline .github/workflows/ci_tests_dev.yaml:32]:
```suggestion
        os: [ubuntu-22.04, macOS-12, windows-2022]
```

**Comment by weiji14** [comment]
@seisman, seems to be working now. Just realized at https://github.com/GenericMappingTools/pygmt/pull/1918#discussion_r897143178 that it was a typo :sweat_smile: Should be `macos-12`, not `macos-12.0` :shrug:

**Review by seisman — APPROVED** [review]

**Pull request merged**